pkg/service/kubernetes_auth.go

```diff
@@ -60,14 +65,20 @@ func (k *KubernetesAuth) JWTAuth(next echo.HandlerFunc) echo.HandlerFunc {
 			token = t
 		}
 
-		client, err := k.CreateClientFunc(token)
-		if err != nil {
-			return err
+		cachedClient, exists := k.cache.Get(token)
+		if !exists {
+			var err error
+			cachedClient, err = k.CreateClientFunc(token)
+			if err != nil {
+				return err
+			}
+			k.cache.Add(token, cachedClient)
 		}
 
 		apiContext := &APIContext{
 			Context: c,
-			client:  client,
+			context: context.TODO(),
```

Comment:
Is this context used somewhere?
I think we can remove it from the `APIContext` struct.

Reply:
Yes, it's used in all client.List/Get/Create calls etc.

Reply:
The alternative would be to set `context.TODO()` for these calls individually and not store it in the APIContext property

Reply:
Hmm I see. I guess we should use `c.Request().Context()` instead...

But I'm fine with doing it in another PR...

Reply:
ah, was searching for something like this, didn't know it's hidden in Request().

A bit of search/replace was easy.